openmc/checkvalue.py

```diff
@@ -16,20 +30,96 @@ def check_type(name, value, expected_type, expected_iter_type=None):
 
     """
 
-    if not isinstance(value, expected_type):
+    if not _isinstance(value, expected_type):
         msg = 'Unable to set "{0}" to "{1}" which is not of type "{2}"'.format(
             name, value, expected_type.__name__)
         raise ValueError(msg)
 
     if expected_iter_type:
         for item in value:
-            if not isinstance(item, expected_iter_type):
+            if not _isinstance(item, expected_iter_type):
                 msg = 'Unable to set "{0}" to "{1}" since each item must be ' \
                       'of type "{2}"'.format(name, value,
                                            expected_iter_type.__name__)
                 raise ValueError(msg)
 
 
+def check_iterable_type(name, value, expected_type, min_depth=1, max_depth=1):
+    """Ensure that an object is an iterable containing an expected type.
+
+    Parameters
+    ----------
+    name : str
+        Description of value being checked
+    value : Iterable
+        Iterable, possibly of other iterables, that should ultimately contain
+        the expected type
+    expected_type : type
+        type that the iterable should contain
+    min_depth : int
+        The minimum number of layers of nested iterables there should be before
+        reaching the ultimately contained items
+    max_depth : int
+        The maximum number of layers of nested iterables ...
+    """
+    # Initialize the tree at the very first item.
+    tree = [value]
+    index = [0]
+
+    # Traverse the tree.
+    while index[0] != len(tree[0]):
+        # If we are done with this level of the tree, go to the next branch on
+        # the level above this one.
+        if index[-1] == len(tree[-1]):
+            del index[-1]
+            del tree[-1]
+            index[-1] += 1
+            continue
+
+        # Get a string representation of the current index in case we raise an
+        # exception.
+        form = '[' + '{:d}, '*(len(index)-1) + '{:d}]'
+        ind_str = form.format(*index)
+
+        # What is the current item we are looking at?
+        current_item = tree[-1][index[-1]]
+
+        # If this item is of the expected type, then we've reached the bottom
+        # level of this branch.
+        if _isinstance(current_item, expected_type):
+            # Is this deep enough?
+            if len(tree) < min_depth:
+                msg = 'Error setting "{0}": The item at {1} does not meet the '\
```

Comment:
I don't think the quotation marks would add any clarity here.

`The item at "[0]" does not meet the minimum depth of "2"`
looks cheesy and no more helpful than
`The item at [0] does not meet the minimum depth of 2`